src/backend/tests/c-lisp/prime.sexp

```diff
@@ -0,0 +1,49 @@
+(c-lisp
+    (define ((print int) (n int)))
+    (define ((square int) (i int))
+        (ret (mul i i)))
+
+    (define ((gcd int) (a int) (b int))
+        (declare (big int))
+        (declare (small int))
+
+        (if (gt a b)
+            ((set big a)
+             (set small b))
+
+             (if (lt a b)
+                ((set small a)
+                 (set big b))
+
+                (ret a)))
+
+        (ret (call gcd small (sub big small))))
+
+    (define ((is-prime bool) (n int))
+        (declare (i int))
+        (declare (iter int))
+        (declare (not-prime bool))
+        (set not-prime #f)
+
+        (for (set i 2) (le (call square i) n) (set i (add i 1))
+            ((set iter i)
+             (set not-prime
+                 (ne (call gcd i n) 1))
+             (if not-prime
```

Comment:
Do syntax check that this is indeed a expression and not a statement. Throw an error if not.